server/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java

```diff
@@ -140,6 +141,17 @@ public static void parseTextField(TextFieldMapper.Builder builder,
             if ("norms".equals(propName)) {
                 parseNorms(builder, name, propNode);
                 iterator.remove();
+            } else if (propName.equals("sources")) {
+                List<String> sources = new ArrayList<>();
+                if (propNode instanceof List<?> nodes) {
+                    for (Object node : nodes) {
+                        sources.add(nodeStringValue(node, null));
+                    }
+                } else {
+                    sources.add(nodeStringValue(propNode, null));
+                }
```

Comment:
Why can this be both either a list or a string? Shouldn't this always be a list?

Reply:
I took a look how copyTo is parsed 
https://github.com/crate/crate/blob/master/server/src/main/java/org/elasticsearch/index/mapper/TypeParsers.java#L229-L237
perhaps it's just paranoid defensive code, idk, I kept it just in case.

Reply:
Maybe copy_to supported both formats. Please remove it here and assert that it's a list. If it isn't we messed up the format somewhere else. Leniency here is hurting more as it creates ambiguity about the actual format and can hide errors.